packages/devtools_app/lib/src/shared/table.dart

```diff
@@ -322,11 +316,13 @@ class Selection<T> {
   Selection({
     this.node,
     this.nodeIndex,
+    this.nodeIndexCalculator,
     this.scrollIntoView = false,
   });
 
   final T? node;
   final int? nodeIndex;
+  final int Function(T)? nodeIndexCalculator;
```

Comment:
do we need both of these? can you add a comment explaining which will be used if they are both specified? should we assert that only one of them is non-null?

Reply:
Just added a TODO to address that having both is probably not necessary. And added assertion so that only one will be specified.